### PR TITLE
editor only loaded when switch to editor first time

### PR DESCRIPTION
**Pull request by theanuradha**
as most cases user may not switch to editor at all. So better to only load first time they request to do so.

### CODE REVIEW

File: info.limpet.stackedcharts.core/src/info/limpet/stackedcharts/core/view/StackedChartsView.java [hunk 1]
```diff
@@ -49,7 +50,9 @@
   // effects
   protected TransitionManager transitionManager = null;
   private Composite chartHolder;
-  private StackedchartsEditControl chartEditor;
+  private Composite editorHolder;
+  private ChartSet charts;
+  private AtomicBoolean initEditor = new AtomicBoolean(true);
```
Comment (IanMayo) [inline info.limpet.stackedcharts.core/src/info/limpet/stackedcharts/core/view/StackedChartsView.java:55]: Hi there @theanuradha 

I haven't encountered **AtomicBoolean** before.  Is it a defence against threading issues?

But, isn't **boolean** thread-safe?

**Comment by IanMayo** [comment]
@theanuradha - did you confirm that the edit pane introduces a delay?

File: info.limpet.stackedcharts.core/src/info/limpet/stackedcharts/core/view/StackedChartsView.java (hunk 1 above)
Comment (theanuradha) [inline info.limpet.stackedcharts.core/src/info/limpet/stackedcharts/core/view/StackedChartsView.java:55]: AtomicBoolean is thread safe-option.

Comment (theanuradha) [inline info.limpet.stackedcharts.core/src/info/limpet/stackedcharts/core/view/StackedChartsView.java:55]: U can not do **getAndSet()** as atomic operation with **volatile** boolean (unless of course you synchronize it).

**Comment by theanuradha** [comment]
@IanMayo  Yes edit has slight delay. and most cases editor will not use by user so no point **init** editor until we really need it.

File: info.limpet.stackedcharts.core/src/info/limpet/stackedcharts/core/view/StackedChartsView.java (hunk 1 above)
Comment (IanMayo) [inline info.limpet.stackedcharts.core/src/info/limpet/stackedcharts/core/view/StackedChartsView.java:55]: Aah, yes - you're performing two steps, but you want them to be treated as a single change.  I see.

**Pull request merged**